### ISSUE TITLE
Segmentation loss becomes nan

### ISSUE THREAD

**Issue by shujonnaha** [arslan-chaudhry/dcsp_segmentation#1]
Hi
The segmentation loss before adapt (i == 0) becomes nan after some steps.

**Comment by arslan-chaudhry** [comment]
Can you please paste the logs? Were you able to train the attention network properly? How are you generating the saliency masks?

**Comment by shujonnaha** [comment]
I think the problem was with the learning rate. While training the segmentation network before adapt, I started with 1e-3 and then after 14 steps the loss became nan. Now I am using lr = 5e-4 and the loss is going down smoothly. I was wondering what learning rate you used while training the segmentation network, not the initial attention network, which worked fine for 1e-3

**Comment by arslan-chaudhry** [comment]
Yes, we lowered the learning rate while training the segmentation network. 5e-4 should work fine.

**Comment by arslan-chaudhry** [comment]
@shujonnaha if you are able to solve the issue, may I please ask you to close this?
